src/bot/commands/RemoveCommand.ts

```diff
@@ -5,16 +5,32 @@ import { Message, Permissions } from 'discord.js';
 import * as sounds from '@util/db/Sounds';
 import localize from '@util/i18n/localize';
 import { existsSound, getPathForSound } from '@util/SoundUtil';
+import Config from '@config/Config';
 import Command from './base/Command';
 
 export default class RemoveCommand implements Command {
   public readonly TRIGGERS = ['remove'];
   public readonly NUMBER_OF_PARAMETERS = 1;
   public readonly USAGE = 'Usage: !remove <sound>';
 
+  private readonly config: Config;
+
+  constructor(config: Config) {
+    this.config = config;
+  }
+
   public run(message: Message, params: string[]) {
     if (!message.member) return;
-    if (!message.member.hasPermission(Permissions.FLAGS.ADMINISTRATOR!)) return;
+
+    let allowedToRemove = false;
+
+    if (this.config.rolesAllowedToRemoveSounds !== undefined) {
+      allowedToRemove = message.member!.roles.cache.some(
+        r => this.config.rolesAllowedToRemoveSounds.indexOf(r.name) >= 0
+      );
+    }
+
+    if (!message.member.hasPermission(Permissions.FLAGS.ADMINISTRATOR!) && !allowedToRemove) return;
```

Comment:
There is multiple things here.

The way you implemented it, the user would have to be in the `rolesAllowedToRemoveSounds` AND they would still need to have the administrator flag. That does not change the functionality all too much, and maybe that is a bit too overkill. We are allowing users to set the roles explicitly, so I don't think we need the extra check anymore. What do you think?

You can use `Array.includes` instead of `.indexOf`.

I don't think you need the bang `!` after `message.member`, as the first line of the method checks for it already. With the `!` you tell TypeScript to not care about nullable values any more, but we already made the explicit check for it.

You need to add `rolesAllowedToRemoveSounds` to both `Config` (see other examples), `ConfigInterface` and `DefaultConfig`. After that it cannot be `undefined` anymore, so the check becomes obsolete.

This is one of the commands that need elevated rights. The same check for the `ADMINISTRATOR` flag, seems to appear 6 times overall in the code base, so I expect us to make the same change across the whole code base. Now, we also don't want to repeat ourselves all too much, right? So if the code needed to check for the rights is a couple lines (like here), we can put it in a different file and call that piece of code instead (e.g. `hasElevatedRights(member)`, as well as re-use it across the other 5 occurrences.

Reply:
I think I applied all of your comments, and sorry, I'm terribly new and bad to typescript 🤦‍♂️